From 610be9f4e8200cc89e04a66ef9665e2721ea3e90 Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Tue, 10 Dec 2024 16:49:54 +0200
Subject: [PATCH] [SW-210370] preboot rtos: integrate common fit utils

Added fit_utils.c which is common among eROM, zephyr-plugins,
zephyr-apps/preboot.

Integrating fit_utils into zephyr-apps/preboot and removed the
redundant functions.

Change-Id: I633e6287278e745e9db720b14a55471d61b7078b
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d980602..439aeaa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -115,6 +115,7 @@
 zephyr_library_sources(emb_common_mod/lib/common/lib_common.c)
 zephyr_library_sources(emb_common_mod/lib/common/boot_desc.c)
 zephyr_library_sources(emb_common_mod/lib/common/pci.c)
+zephyr_library_sources(emb_common_mod/lib/common/fit_utils.c)
 zephyr_library_sources(emb_common_mod/lib/preboot/fit.c)
 zephyr_library_sources(emb_common_mod/lib/preboot/fit_rsa.c)
 zephyr_library_sources(emb_common_mod/lib/preboot/fit_regions.c)
diff --git a/include/preboot.h b/include/preboot.h
index 297f002..3fa8ce4 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -66,13 +66,6 @@
 	PREBOOT_EFUSE_VER,
 };
 
-enum fit_sec_failure {
-	FIT_SEC_NONE = 0,
-	FIT_SEC_SVN_INVALID,
-	FIT_SEC_KEY_REVOKED,
-	FIT_SEC_SIGN_FAILED,
-};
-
 struct hl_nic_en {
 	bool is_valid;
 	uint32_t bitmask;
@@ -200,9 +193,6 @@
 void preboot_finalize_rma_tables(void);
 
 bool preboot_is_image_sig_supported(void);
-int preboot_verify_img_sig(const void *fit, size_t img_size,
-			   enum fit_sec_failure *res, u32 *key_idx);
-size_t preboot_get_img_sec_desc_size(void);
 bool preboot_check_bis_compliant_mode(void);
 
 u32 preboot_error_read_and_clr(u32 *major, u32 *minor);
diff --git a/lib/module/module_lkd.c b/lib/module/module_lkd.c
index 5479d50..2256e43 100644
--- a/lib/module/module_lkd.c
+++ b/lib/module/module_lkd.c
@@ -218,7 +218,7 @@
 	int err;
 	const void *data = 0;
 
-	err = fit_image_get_loadable_data(fit, &data, sizeof(u32));
+	err = fit_get_loadable_image_data(fit, &data, sizeof(u32));
 	if (!err)
 		err = preboot_set_sram_sf_access(*(u32 *)data);
 
